exonum/src/node/consensus.rs

```diff
@@ -579,30 +579,23 @@ impl NodeHandler {
 
     /// Handles external boxed transaction. Additionally transaction will be broadcast to the
     /// Node's peers.
+    #[cfg_attr(feature = "cargo-clippy", allow(needless_pass_by_value))]
     pub fn handle_incoming_tx(&mut self, msg: Box<Transaction>) {
```

Comment:
This will definitely break the revertible database in the testkit - it assumes that each merge corresponds to a new block. I think a good solution would be to strongly separate non-consensus and consensus storage interfaces (we've discussed it a bit previously).